image_captioning/models/InceptionV3Encoder.py

```diff
@@ -8,24 +8,29 @@ class InceptionV3Encoder(tf.keras.Model):
     """Encoder using the Inception V3 backbone
     """
 
-    def __init__(self, embedding_dim: int):
+    def __init__(self, embedding_dim: int, train_backbone: bool = False):
         """
         Arguments:
             embedding_dim: The number of channels to embed the output into.
+            train_backbone: Whether to make backbone weights trainable
         """
         super().__init__()
         self.embedding_dim = embedding_dim
 
         # Using Inception V3 backbone
         self.backbone = tf.keras.applications.InceptionV3(include_top=False,
                                                           weights='imagenet')
+        if not train_backbone:
+            for layer in self.backbone.layers:
+                layer.trainable = False
```

Comment:
Isn't `self.backbone.trainable = train_backbone` enough to set trainability? I was referring [this example](https://www.tensorflow.org/guide/keras/transfer_learning#the_typical_transfer-learning_workflow) and it seems it sets `trainable = False` only to the outer model.

Reply:
lemme check

Reply:
I think that's right. Lemme change it for simplicity's sake.